File: backend/app/schemas/messaging.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import MessageType


class CounterpartRead(BaseModel):
    user_id: uuid.UUID
    name: str
    picture_url: str | None


class MessageRead(BaseModel):
    id: uuid.UUID
    conversation_id: uuid.UUID
    sender_user_id: uuid.UUID
    is_mine: bool
    message_type: MessageType
    text_body: str | None
    attachment_url: str | None
    attachment_original_filename: str | None
    attachment_mime_type: str | None
    attachment_size_bytes: int | None
    created_at: datetime
    read_at: datetime | None


class ConversationRead(BaseModel):
    id: uuid.UUID
    counterpart: CounterpartRead
    last_message: MessageRead | None
    unread_count: int
    last_message_at: datetime


class StartConversationRequest(BaseModel):
    counterpart_id: uuid.UUID


class SendTextMessageRequest(BaseModel):
    text: str = Field(min_length=1, max_length=4000)

File: frontend/src/types/messaging.ts
export type MessageType = "text" | "audio" | "video" | "voice_note" | "document";

export interface Counterpart {
  user_id: string;
  name: string;
  picture_url: string | null;
}

export interface Message {
  id: string;
  conversation_id: string;
  sender_user_id: string;
  is_mine: boolean;
  message_type: MessageType;
  text_body: string | null;
  attachment_url: string | null;
  attachment_original_filename: string | null;
  attachment_mime_type: string | null;
  attachment_size_bytes: number | null;
  created_at: string;
  read_at: string | null;
}

export interface Conversation {
  id: string;
  counterpart: Counterpart;
  last_message: Message | null;
  unread_count: number;
  last_message_at: string;
}
